src/suggestion-box.ts

```diff
@@ -0,0 +1,409 @@
+            this.size = suggestions.length;
+            this.filtered_list = suggestions;
+            this.show_container = true;
+            for (let i of suggestions) {
+                let el = document.createElement("div");
+                el.className = "sgn-item";
+                el.slot = "__private-item";
+                console.log("key value ", i);
+                el.innerHTML =
+                    i.html ||
+                    (this.template_slots[0] == null || undefined
+                        ? this.highlight == "matched"
+                            ? this._startsWithHighlight(i[key])
+                            : this._highlight(
+                                  i[key],
+                                  (this.input_el as HTMLInputElement)?.value
+                              )
+                        : this._format(
+                              this.template_slots[0].innerHTML,
+                              i,
+                              key
+                          ));
+                this.appendChild(el);
+            }
+            if (has_more) {
+                let el = document.createElement("div");
+                el.className = "sgn-item";
+                el.slot = "__private-item";
+
+                el.innerHTML = `<div class="sgn-loadmore"><div>Loadmore</div></div>`;
+                this.appendChild(el);
+            }
+            return this;
+        } else {
+            if (this.private_items?.length) {
+                for (let el of this.private_items) el.remove();
+            }
+            return;
+        }
+    }
+
+    private _loadmoreEntry(event: Event) {
+        const loadmore = new CustomEvent("loadmore", {
+            bubbles: true,
+        });
+        if (true) {
+            this.private_items[this.private_items.length - 1].remove();
+        }
+
+        this.dispatchEvent(loadmore);
+        event.stopPropagation();
+    }
+
+    appendSuggestion(
+        suggestions: Array<{ [key: string]: string }>,
+        has_more: boolean,
+        key: string
+    ) {
+        this.result = [...this.result];
+        this.result.push(...suggestions);
+        this.has_more = has_more || false;
+        this.size = this.size + suggestions.length;
+
+        for (let i of suggestions) {
+            let el = document.createElement("div");
+            el.className = "sgn-item";
+            el.slot = "__private-item";
+            el.innerHTML =
+                i.html ||
+                (this.template_slots[0] == null || undefined
+                    ? this._startsWithHighlight(i[key])
+                    : this._format(this.template_slots[0].innerHTML, i, key));
+            this.appendChild(el);
+        }
+
+        if (has_more) {
+            let el = document.createElement("div");
+            el.className = "sgn-item";
+            el.slot = "__private-item";
+            el.innerHTML = `<div>Loadmore</div>`;
+            this.appendChild(el);
+        }
```

Comment:
I think this block is very much similar to the code in the setSuggestions. We can avoid redundancy by putting the common code in a function and calling that function.